jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/messages/ByteArrayMessageSink.java

```diff
@@ -97,14 +102,14 @@ public void accept(Frame frame, Callback callback)
         }
     }
 
-    private void aggregatePayload(Frame frame) throws IOException
+    private void aggregatePayload(Frame frame, Callback callback)
```

Comment:
this method feels verbose and like it is double checking stuff.   I think if you inline it then the hasPayload stuff will already have been done.

Reply:
It's not double checking anything but I can inline it.